worker/interactions.go

```diff
@@ -4,26 +4,52 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"sync"
 	"time"
 
 	"go.sia.tech/jape"
 	"go.sia.tech/renterd/hostdb"
+	"go.uber.org/zap"
 )
 
 const (
 	keyInteractionRecorder contextKey = "InteractionRecorder"
 )
 
 type (
-	InteractionRecorder interface {
+	HostInteractionRecorder interface {
 		RecordHostScan(...hostdb.HostScan)
 		RecordPriceTableUpdate(...hostdb.PriceTableUpdate)
+		Stop(context.Context)
 	}
+
+	hostInteractionRecorder struct {
+		bus    Bus
+		logger *zap.SugaredLogger
+
+		mu            sync.Mutex
+		flushInterval time.Duration
```

Comment:
Another thought for post-rhpv4 is getting rid of this buffer altogether and batch the writes on the bus side.

The worker would just launch a goroutine that sends an interaction to the bus and have a `sync.WaitGroup` to make sure we don't shut down before all those goroutines are done.

Reply:
yeah buffering on the `bus` is better, I think there was a reason we had it on the worker though 🤔 , but yeah should move to the `bus`